ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt

```diff
@@ -2,17 +2,69 @@
  * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
+@file:Suppress("NOTHING_TO_INLINE", "unused")
+
 package io.ktor.network.quic.bytes
 
 import io.ktor.utils.io.core.*
 
+internal typealias UInt8 = UByte
+internal typealias UInt16 = UShort
+internal typealias UInt24 = UInt
+internal typealias UInt32 = UInt
+internal typealias UInt64 = ULong
+
+internal inline fun UInt8.toUInt64(): UInt64 = toULong()
+internal inline fun UInt16.toUInt64(): UInt64 = toULong()
+internal inline fun UInt32.toUInt64(): UInt64 = toULong()
+internal inline fun UInt8.toUInt32(): UInt32 = toUInt()
+internal inline fun UInt16.toUInt32(): UInt32 = toUInt()
+internal inline fun UInt8.toUInt16(): UInt16 = toUShort()
+
+internal inline fun UInt64.toUInt8(): UInt8 = toUByte()
+internal inline fun UInt32.toUInt8(): UInt8 = toUByte()
+internal inline fun UInt16.toUInt8(): UInt8 = toUByte()
+internal inline fun UInt64.toUInt16(): UInt16 = toUShort()
+internal inline fun UInt32.toUInt16(): UInt16 = toUShort()
+internal inline fun UInt64.toUInt32(): UInt32 = toUInt()
+
+@OptIn(ExperimentalUnsignedTypes::class)
```

Comment:
Could be replaced with file:
@file:OptIn(ExperimentalUnsignedTypes::class)

Reply:
I'm not sure about compiling, but IDE shows me that this approach does not work (it still requires opt-in for separate functions)

Reply:
checked, guess it's IDE bug